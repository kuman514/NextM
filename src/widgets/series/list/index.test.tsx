import { render } from '@testing-library/react';

import SeriesList from '.';

jest.mock(
  'next/link',
  () =>
    function MockedLink(props: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
      return <a {...props} />;
    }
);

describe('SeriesList', () => {
  it('should render as expected', () => {
    const { container } = render(
      <SeriesList
        label="Series"
        seriesDataList={[
          {
            id: 1,
            title: 'Spiderman',
            description: '',
            modified: new Date(),
            startYear: 1980,
            endYear: 2099,
            rating: 'pg-13',
            thumbnail: {
              path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
              extension: 'jpg',
            },
            resourceURI: '',
            characters: {
              available: 0,
              collectionURI: '',
              items: [],
              returned: 0,
            },
            stories: {
              available: 0,
              collectionURI: '',
              items: [],
              returned: 0,
            },
            comics: {
              available: 0,
              collectionURI: '',
              items: [],
              returned: 0,
            },
            next: {
              resourceURI: '',
              name: '',
            },
            previous: {
              resourceURI: '',
              name: '',
            },
            urls: [],
          },
          {
            id: 2,
            title: 'Avengers',
            description: '',
            modified: new Date(),
            startYear: 2012,
            endYear: 2019,
            rating: 'pg-13',
            thumbnail: {
              path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
              extension: 'jpg',
            },
            resourceURI: '',
            characters: {
              available: 0,
              collectionURI: '',
              items: [],
              returned: 0,
            },
            stories: {
              available: 0,
              collectionURI: '',
              items: [],
              returned: 0,
            },
            comics: {
              available: 0,
              collectionURI: '',
              items: [],
              returned: 0,
            },
            next: {
              resourceURI: '',
              name: '',
            },
            previous: {
              resourceURI: '',
              name: '',
            },
            urls: [],
          },
        ]}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
