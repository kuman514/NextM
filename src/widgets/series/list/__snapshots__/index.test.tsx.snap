// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesList should render as expected 1`] = `
<div
  class="w-full flex flex-col items-start"
>
  <h1
    class="mt-9 mb-6 text-4xl font-bold undefined"
  >
    Series
  </h1>
  <ul
    class="w-full flex flex-row overflow-x-auto gap-4"
  >
    <li>
      <a
        aria-label="list-item-link"
        class="flex flex-col w-48 h-72 justify-center rounded-xl overflow-hidden bg-black hover:bg-red-700 transition"
        href="/series/1"
      >
        <div
          class="w-48 h-48"
        >
          <img
            alt="Spiderman"
            class="w-full h-full object-cover"
            data-nimg="1"
            decoding="async"
            height="192"
            loading="lazy"
            src="/_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=384&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="192"
          />
        </div>
        <div
          class="flex flex-col justify-center items-center w-full h-36"
        >
          <span
            class="text-white text-center"
          >
            Spiderman
          </span>
          <span
            class="text-white text-center"
          >
            1980 ~
          </span>
        </div>
      </a>
    </li>
    <li>
      <a
        aria-label="list-item-link"
        class="flex flex-col w-48 h-72 justify-center rounded-xl overflow-hidden bg-black hover:bg-red-700 transition"
        href="/series/2"
      >
        <div
          class="w-48 h-48"
        >
          <img
            alt="Avengers"
            class="w-full h-full object-cover"
            data-nimg="1"
            decoding="async"
            height="192"
            loading="lazy"
            src="/_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=384&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="192"
          />
        </div>
        <div
          class="flex flex-col justify-center items-center w-full h-36"
        >
          <span
            class="text-white text-center"
          >
            Avengers
          </span>
          <span
            class="text-white text-center"
          >
            2012 ~ 2019
          </span>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
